.githooks/pre-commit

```diff
@@ -21,12 +22,23 @@ go fmt $(go list ./... | grep -v /vendor/)
 
 # Check all files for suspicious constructs
 {
-	go vet $(go list ./... | grep -v /vendor/)
+	go vet ${FILES}
 } || {
 	exitStatus=$?
 
 	if [ $exitStatus ]; then
 		printf "\nIssues found in your code, please fix them and try again."
 		exit 1
 	fi
+}
```

Comment:
Should the test run every time there is a commit , if so its less like a test and more like production